src/shared/localShadowRepo.ts

```diff
@@ -294,14 +302,26 @@ export class ShadowRepo {
       }
     }
 
-    for (const filepath of [...new Set(this.isWindows ? deletedFiles.map(normalize).map(ensurePosix) : deletedFiles)]) {
-      try {
-        // these need to be done sequentially because isogit manages file locking.  Isogit remove does not support multiple files at once
-        // eslint-disable-next-line no-await-in-loop
-        await git.remove({ fs, dir: this.projectPath, gitdir: this.gitDir, filepath });
-      } catch (e) {
-        redirectToCliRepoError(e);
+    if (deletedFiles.length) {
+      // Using a cache here speeds up the performance by ~24.4%
+      let cache = {};
+      const deleteMarker = Performance.mark('@salesforce/source-tracking', 'localShadowRepo.commitChanges#delete', {
+        deletedFiles: deletedFiles.length,
+      });
+      for (const filepath of [
+        ...new Set(this.isWindows ? deletedFiles.map(normalize).map(ensurePosix) : deletedFiles),
+      ]) {
+        try {
+          // these need to be done sequentially because isogit manages file locking.  Isogit remove does not support multiple files at once
+          // eslint-disable-next-line no-await-in-loop
+          await git.remove({ fs, dir: this.projectPath, gitdir: this.gitDir, filepath, cache });
+        } catch (e) {
+          redirectToCliRepoError(e);
+        }
       }
+      // clear cache
```

Comment:
what problem was that causing?

Reply:
I wasn't seeing a problem with this. It is recommended by isogit so that it is garbage collected
https://isomorphic-git.org/docs/en/cache#the-cache-parameter